roachpb: TestMakePriority failed

roachpb.TestMakePriority [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13727216?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/13727216?buildTab=artifacts#/) on release-23.2 @ [674f8f48764748a3a98771c62f236d2454e0d438](https://github.com/cockroachdb/cockroach/commits/674f8f48764748a3a98771c62f236d2454e0d438):


```
=== RUN   TestMakePriority
    data_test.go:978: 0.001100 vs 0.010000: 74078 wins, 675922 losses (winRatio=0.109595, priRatio=0.110000)
    data_test.go:978: 0.001100 vs 0.100000: 8141 wins, 741859 losses (winRatio=0.010974, priRatio=0.011000)
    data_test.go:978: 0.001100 vs 0.500000: 1611 wins, 748389 losses (winRatio=0.002153, priRatio=0.002200)
    data_test.go:978: 0.001100 vs 1.000000: 818 wins, 749182 losses (winRatio=0.001092, priRatio=0.001100)
    data_test.go:978: 0.001100 vs 1.000000: 760 wins, 749240 losses (winRatio=0.001014, priRatio=0.001100)
    data_test.go:978: 0.010000 vs 0.100000: 67527 wins, 682473 losses (winRatio=0.098945, priRatio=0.100000)
    data_test.go:978: 0.010000 vs 0.500000: 14638 wins, 735362 losses (winRatio=0.019906, priRatio=0.020000)
    data_test.go:978: 0.010000 vs 1.000000: 7417 wins, 742583 losses (winRatio=0.009988, priRatio=0.010000)
    data_test.go:978: 0.010000 vs 1.000000: 7422 wins, 742578 losses (winRatio=0.009995, priRatio=0.010000)
    data_test.go:978: 0.010000 vs 2.000000: 3622 wins, 746378 losses (winRatio=0.004853, priRatio=0.005000)
    data_test.go:978: 0.100000 vs 0.500000: 124867 wins, 625133 losses (winRatio=0.199745, priRatio=0.200000)
    data_test.go:978: 0.100000 vs 1.000000: 68530 wins, 681470 losses (winRatio=0.100562, priRatio=0.100000)
    data_test.go:978: 0.100000 vs 1.000000: 68322 wins, 681678 losses (winRatio=0.100226, priRatio=0.100000)
    data_test.go:978: 0.100000 vs 2.000000: 35440 wins, 714560 losses (winRatio=0.049597, priRatio=0.050000)
    data_test.go:978: 0.100000 vs 10.000000: 7523 wins, 742477 losses (winRatio=0.010132, priRatio=0.010000)
    data_test.go:978: 0.500000 vs 1.000000: 249741 wins, 500259 losses (winRatio=0.499223, priRatio=0.500000)
    data_test.go:978: 0.500000 vs 1.000000: 249404 wins, 500596 losses (winRatio=0.498214, priRatio=0.500000)
    data_test.go:978: 0.500000 vs 2.000000: 149878 wins, 600122 losses (winRatio=0.249746, priRatio=0.250000)
    data_test.go:978: 0.500000 vs 10.000000: 35515 wins, 714485 losses (winRatio=0.049707, priRatio=0.050000)
    data_test.go:978: 0.500000 vs 100.000000: 3762 wins, 746238 losses (winRatio=0.005041, priRatio=0.005000)
    data_test.go:978: 1.000000 vs 1.000000: 374384 wins, 375616 losses (winRatio=0.996720, priRatio=1.000000)
    data_test.go:978: 1.000000 vs 2.000000: 249569 wins, 500431 losses (winRatio=0.498708, priRatio=0.500000)
    data_test.go:978: 1.000000 vs 10.000000: 68198 wins, 681802 losses (winRatio=0.100026, priRatio=0.100000)
    data_test.go:978: 1.000000 vs 100.000000: 7463 wins, 742537 losses (winRatio=0.010051, priRatio=0.010000)
    data_test.go:978: 1.000000 vs 999.000000: 826 wins, 749174 losses (winRatio=0.001103, priRatio=0.001001)
    data_test.go:978: 1.000000 vs 2.000000: 250317 wins, 499683 losses (winRatio=0.500952, priRatio=0.500000)
    data_test.go:978: 1.000000 vs 10.000000: 68158 wins, 681842 losses (winRatio=0.099962, priRatio=0.100000)
    data_test.go:978: 1.000000 vs 100.000000: 7548 wins, 742452 losses (winRatio=0.010166, priRatio=0.010000)
    data_test.go:978: 1.000000 vs 999.000000: 910 wins, 749090 losses (winRatio=0.001215, priRatio=0.001001)
    data_test.go:981: Error (1.000000 vs 999.000000: 910 wins, 749090 losses): winRatio=0.001215 not close to priRatio=0.001001
    data_test.go:978: 2.000000 vs 10.000000: 125400 wins, 624600 losses (winRatio=0.200768, priRatio=0.200000)
    data_test.go:978: 2.000000 vs 100.000000: 14764 wins, 735236 losses (winRatio=0.020081, priRatio=0.020000)
    data_test.go:978: 2.000000 vs 999.000000: 1584 wins, 748416 losses (winRatio=0.002116, priRatio=0.002002)
    data_test.go:978: 10.000000 vs 100.000000: 68387 wins, 681613 losses (winRatio=0.100331, priRatio=0.100000)
    data_test.go:978: 10.000000 vs 999.000000: 7508 wins, 742492 losses (winRatio=0.010112, priRatio=0.010010)
    data_test.go:978: 100.000000 vs 999.000000: 68362 wins, 681638 losses (winRatio=0.100291, priRatio=0.100100)
--- FAIL: TestMakePriority (2.44s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMakePriority.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35718

Off by 0.2%:

```
    data_test.go:981: Error (1.000000 vs 999.000000: 910 wins, 749090 losses): winRatio=0.001215 not close to priRatio=0.001001
```

Unable to repro over 30k runs.

--- a/pkg/roachpb/data_test.go
+++ b/pkg/roachpb/data_test.go
@@ -929,7 +929,7 @@ func TestMakePriority(t *testing.T) {
 	}
 
 	// Generate values for all priorities.
-	const trials = 750000
+	const trials = 1000000
 	values := make([][trials]enginepb.TxnPriority, len(userPs))
 	for i, userPri := range userPs {
 		for tr := 0; tr < trials; tr++ {

